validations/src/validations.rs

```diff
@@ -264,66 +263,74 @@ pub fn evaluate_tally_precondition_clause(
 
     // Count how many times is each RADON type featured in `reveals`, but count `RadonError` items
     // separately as they need to be handled differently.
-    let reveals_len = reveals.len();
-    let mut values_counter = Counter::new(RadonTypes::num_types());
-    let mut errors_counter = 0u32;
+    let reveals_len = reveals.len() as u32;
+    let mut counter = Counter::new(RadonTypes::num_types());
     for reveal in &reveals {
-        match reveal.result {
-            RadonTypes::RadonError(_) => errors_counter += 1,
-            _ => values_counter.increment(reveal.result.discriminant()),
-        }
+        counter.increment(reveal.result.discriminant());
     }
 
-    // Compute ratio of reveals that are errors
-    let error_ratio = errors_counter as f64 / reveals_len as f64;
-
-    // If the ratio of errors is over the user-defined threshold, return the mode of the errors.
-    // Otherwise, return only non-error reveals, paired with a "liars" vector that positionally
-    // tells which of the input reveals were in consensus with the mode type (most frequent revealed
-    // type after discarding the errors).
-    if error_ratio > non_error_min {
-        let errors: Vec<RadonTypes> = reveals
-            .into_iter()
-            .filter_map(|reveal| match reveal.into_inner() {
-                radon_types @ RadonTypes::RadonError(_) => Some(radon_types),
-                _ => None,
-            })
-            .collect();
+    // Compute ratio of type consensus amongst reveals (percentage of reveals that have same type
+    // as the frequent type).
+    let achieved_consensus = counter.max_val as f64 / reveals_len as f64;
 
-        let errors_array = RadonArray::from(errors);
+    // If the achieved consensus is over the user-defined threshold, continue.
+    // Otherwise, return `RadError::InsufficientConsensus`.
+    if achieved_consensus > non_error_min {
```

Comment:
`non_error_min` is now misleading, it should be named `consensus_min` or something.